drivers/wifi/esp32/src/esp_wifi_drv.c

```diff
@@ -182,6 +182,12 @@ static void scan_done_handler(void)
 			strncpy(res.ssid, ap_list_buffer[k].ssid, ssid_len);
 			res.rssi = ap_list_buffer[k].rssi;
 			res.channel = ap_list_buffer[k].primary;
+
+			if (ap_list_buffer[k].bssid) {
```

Comment:
I suggest removing this `if (ap_list_buffer[k].bssid) {`. It will probably drop build warnings due to comparison.

Reply:
Have you checked whether build completes without any warning?

Reply:
I have made the check more explicit. The checkpatch script does not seem to give out any warnings if that's what you meant.